### ISSUE TITLE
Do not flag for exact floating-point numbers

### ISSUE THREAD

**Issue by oscargus** [rust-lang/rust-clippy#11887]
### Description

Sometimes it makes sense to write a floating-point number with a higher precision than is required to uniquely represent it. For example, if we would like to clarify that this it the exact number. An example:
```rust
1 |             let fp = 0.0000152587890625f32 // 0.0000152587890625 = 2^-16
  |                      ^^^^^^^^^^^^^^^^^^^^^ help: consider changing the type or truncating it to: `0.000_015_258_789_f32`
```

I'd suggest that if the number can be represented exactly, leave it as is. While truncating it will still represent the same number, the suggestion to change the type is not relevant as this is exactly a 32-bit floating-point number.

### Version

```text
rustc 1.74.0 (79e9716c9 2023-11-13)
binary: rustc
commit-hash: 79e9716c980570bfd1f666e3b16ac583f0168962
commit-date: 2023-11-13
host: x86_64-unknown-linux-gnu
release: 1.74.0
LLVM version: 17.0.4
```

**Comment by samueltardieu** [comment]
I would even argue that this is a bug, as this lint description states that it "checks for float literals with a precision greater than that supported by the underlying type" which is not the case here.
